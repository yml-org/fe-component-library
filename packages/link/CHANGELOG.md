# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.1.0 (2023-03-13)

### Bug Fixes

- **app:** update package versions ([c7af278](https://github.com/yml-org/fe-component-library/commit/c7af27899d3679b497126fb26cf2ee71920c7fd8))
- **textarea:** fix storybook ([e2a9109](https://github.com/yml-org/fe-component-library/commit/e2a9109798c17c77092555e83d73f50f0aedfec6))
- **version:** update package versions ([2c20a65](https://github.com/yml-org/fe-component-library/commit/2c20a6508da35e35a82b3572c27d37f5828749a6))
- **version:** update package versions ([7d8cf7f](https://github.com/yml-org/fe-component-library/commit/7d8cf7f15d4f548555eb74705c83e8125f89486c))
- **version:** update package versions ([ebeb6fa](https://github.com/yml-org/fe-component-library/commit/ebeb6faaae123cac74215813d1d1ffaf01fee0cc))
- **version:** update package versions ([d0aeb30](https://github.com/yml-org/fe-component-library/commit/d0aeb30a06f38e6a5b5b2f309f2cd86448a848e8))

### Features

- **storybook:** improve documentation ([#24](https://github.com/yml-org/fe-component-library/issues/24)) ([e273d09](https://github.com/yml-org/fe-component-library/commit/e273d09b6eb5fd842f23ea8bcbd56fb160b827e8))

## [0.0.1](https://github.com/yml-org/fe-component-library/compare/@yml-webcl/link@0.2.0...@yml-webcl/link@0.0.1) (2023-03-13)

### Bug Fixes

- **app:** update package versions ([c7af278](https://github.com/yml-org/fe-component-library/commit/c7af27899d3679b497126fb26cf2ee71920c7fd8))

# 0.2.0 (2023-03-13)

### Bug Fixes

- **textarea:** fix storybook ([e2a9109](https://github.com/yml-org/fe-component-library/commit/e2a9109798c17c77092555e83d73f50f0aedfec6))
- **version:** update package versions ([2c20a65](https://github.com/yml-org/fe-component-library/commit/2c20a6508da35e35a82b3572c27d37f5828749a6))
- **version:** update package versions ([7d8cf7f](https://github.com/yml-org/fe-component-library/commit/7d8cf7f15d4f548555eb74705c83e8125f89486c))
- **version:** update package versions ([ebeb6fa](https://github.com/yml-org/fe-component-library/commit/ebeb6faaae123cac74215813d1d1ffaf01fee0cc))
- **version:** update package versions ([d0aeb30](https://github.com/yml-org/fe-component-library/commit/d0aeb30a06f38e6a5b5b2f309f2cd86448a848e8))

### Features

- **storybook:** improve documentation ([#24](https://github.com/yml-org/fe-component-library/issues/24)) ([e273d09](https://github.com/yml-org/fe-component-library/commit/e273d09b6eb5fd842f23ea8bcbd56fb160b827e8))

# 0.1.0 (2023-03-13)

### Bug Fixes

- **textarea:** fix storybook ([e2a9109](https://github.com/yml-org/fe-component-library/commit/e2a9109798c17c77092555e83d73f50f0aedfec6))
- **version:** update package versions ([2c20a65](https://github.com/yml-org/fe-component-library/commit/2c20a6508da35e35a82b3572c27d37f5828749a6))
- **version:** update package versions ([7d8cf7f](https://github.com/yml-org/fe-component-library/commit/7d8cf7f15d4f548555eb74705c83e8125f89486c))
- **version:** update package versions ([ebeb6fa](https://github.com/yml-org/fe-component-library/commit/ebeb6faaae123cac74215813d1d1ffaf01fee0cc))
- **version:** update package versions ([d0aeb30](https://github.com/yml-org/fe-component-library/commit/d0aeb30a06f38e6a5b5b2f309f2cd86448a848e8))

### Features

- **storybook:** improve documentation ([#24](https://github.com/yml-org/fe-component-library/issues/24)) ([e273d09](https://github.com/yml-org/fe-component-library/commit/e273d09b6eb5fd842f23ea8bcbd56fb160b827e8))

# 0.2.0 (2023-03-13)

### Bug Fixes

- **version:** update package versions ([2c20a65](https://github.com/yml-org/fe-component-library/commit/2c20a6508da35e35a82b3572c27d37f5828749a6))
- **version:** update package versions ([7d8cf7f](https://github.com/yml-org/fe-component-library/commit/7d8cf7f15d4f548555eb74705c83e8125f89486c))
- **version:** update package versions ([ebeb6fa](https://github.com/yml-org/fe-component-library/commit/ebeb6faaae123cac74215813d1d1ffaf01fee0cc))
- **version:** update package versions ([d0aeb30](https://github.com/yml-org/fe-component-library/commit/d0aeb30a06f38e6a5b5b2f309f2cd86448a848e8))

### Features

- **storybook:** improve documentation ([#24](https://github.com/yml-org/fe-component-library/issues/24)) ([e273d09](https://github.com/yml-org/fe-component-library/commit/e273d09b6eb5fd842f23ea8bcbd56fb160b827e8))

## [0.1.2](https://github.com/yml-org/fe-component-library/compare/@yml-webcl/link@0.1.1...@yml-webcl/link@0.1.2) (2023-03-13)

### Bug Fixes

- **version:** update package versions ([2c20a65](https://github.com/yml-org/fe-component-library/commit/2c20a6508da35e35a82b3572c27d37f5828749a6))

## [0.1.1](https://github.com/yml-org/fe-component-library/compare/@yml-webcl/link@0.1.0...@yml-webcl/link@0.1.1) (2023-03-13)

### Bug Fixes

- **version:** update package versions ([7d8cf7f](https://github.com/yml-org/fe-component-library/commit/7d8cf7f15d4f548555eb74705c83e8125f89486c))

# 0.1.0 (2023-03-13)

### Bug Fixes

- **version:** update package versions ([ebeb6fa](https://github.com/yml-org/fe-component-library/commit/ebeb6faaae123cac74215813d1d1ffaf01fee0cc))
- **version:** update package versions ([d0aeb30](https://github.com/yml-org/fe-component-library/commit/d0aeb30a06f38e6a5b5b2f309f2cd86448a848e8))

### Features

- **storybook:** improve documentation ([#24](https://github.com/yml-org/fe-component-library/issues/24)) ([e273d09](https://github.com/yml-org/fe-component-library/commit/e273d09b6eb5fd842f23ea8bcbd56fb160b827e8))
