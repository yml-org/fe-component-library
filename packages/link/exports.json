{
  ".": "./src/index.js",
  "./src/*.js": "./src/*.js",
  "./package.json": "./package.json",
  "./link.component.js": "./link.component.js"
}
