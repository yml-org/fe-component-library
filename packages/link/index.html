<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>web components</title>
  </head>
  <body>
    <ymlwebcl-link href="#">
  </ymlwebcl-link>
    <script type="module" src="src/link/link.component.ts"></script>
  </body>
</html>
