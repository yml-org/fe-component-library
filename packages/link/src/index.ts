export * from './link/link.component';
