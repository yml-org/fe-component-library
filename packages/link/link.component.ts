import { LinkComponent } from './src/link/link.component';

declare global {
  interface HTMLElementTagNameMap {
    'ymlwebcl-link': LinkComponent;
  }
}
