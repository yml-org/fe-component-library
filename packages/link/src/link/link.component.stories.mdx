import { html } from 'lit';
import { Story, Meta, Canvas } from '@storybook/addon-docs';
import './link.component';
import {
  linkConfig,
  linkConfigTypeButton,
  linkConfigSlot,
  linkConfigSlotTypeButton,
} from '../constants/link.variants';

<Meta title="Link" />

# Link Component

The link component is a Lit component that renders an anchor element that is framework independent and utilizes Tailwind CSS internally.

# Props

| Prop Name | Type               | Description                                                                                               |
| --------- | ------------------ | --------------------------------------------------------------------------------------------------------- |
| text      | string             | The text for the button element                                                                           |
| linkTitle | string             | The tooltip text on hover                                                                                 |
| href      | string             | The redirection URL                                                                                       |
| slotName  | string             | The name for the slot that can be used to pass children                                                   |
| rel       | string             | The rel attribute which defines the relationship between the current page and the linked page or resource |
| target    | string             | The target attribute that specifies where to open the linked document                                     |
| type      | LinkComponentTypes | The link component type                                                                                   |

The link has the following types:

1. link
2. button

# Examples

Basic link component without `props`

export const TemplateDefault = () => html`<link-component></link-component>`;

<Canvas columns={1}>
  <Story name="default">{TemplateDefault.bind({})}</Story>
</Canvas>

Link component with `props`

export const TemplateProps = (args) => html`<link-component
  .href=${args.linkProps.href}
  .text=${args.linkProps.text}
  .linkTitle=${args.linkProps.linkTitle}
  .rel=${args.linkProps.rel}
  .target=${args.linkProps.target}
  .type=${args.linkProps.type}
>
</link-component>`;

<Canvas columns={1}>
  <Story
    name="withProps"
    args={{
      linkProps: linkConfig,
    }}
  >
    {TemplateProps.bind({})}
  </Story>
</Canvas>

Link component with type `button`

<Canvas columns={1}>
  <Story
    name="typeButton"
    args={{
      linkProps: linkConfigTypeButton,
    }}
  >
    {TemplateProps.bind({})}
  </Story>
</Canvas>

Link component with `slot`

export const TemplateSlot = (args) => html`<link-component
  .href=${args.linkSlot.href}
  .text=${args.linkSlot.text}
  .linkTitle=${args.linkSlot.linkTitle}
  .rel=${args.linkSlot.rel}
  .target=${args.linkSlot.target}
  .type=${args.linkSlot.type}
  .slotName=${args.linkSlot.slotName}
>
  <div slot="newSlot" style="width:auto;  display: inline-block">
    <p>Slot elements wrapped within an anchor tag</p>
    <img
      class="h-9 w-9"
      src="https://avatars.mds.yandex.net/i?id=84dbd50839c3d640ebfc0de20994c30d-4473719-images-taas-consumers&n=27&h=480&w=480"
      alt=""
    />
  </div>
</link-component>`;

<Canvas columns={1}>
  <Story
    name="withSlot"
    args={{
      linkSlot: linkConfigSlot,
    }}
  >
    {TemplateSlot.bind({})}
  </Story>
</Canvas>

Link component with `slot` and type `button`

<Canvas columns={1}>
  <Story
    name="withSlotTypeButton"
    args={{
      linkSlot: linkConfigSlotTypeButton,
    }}
  >
    {TemplateSlot.bind({})}
  </Story>
</Canvas>
