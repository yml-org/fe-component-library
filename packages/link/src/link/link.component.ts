import { html } from 'lit';
import { msg, str } from '@lit/localize';
import { customElement, property } from 'lit/decorators.js';
import { TailwindElement } from '../shared/tailwind.element';
import { LinkTypes } from '../constants/link.component';
import { LinkComponentTypes } from '../types/link.component';

@customElement('link-component')
export class LinkComponent extends TailwindElement(null) {
  @property()
  text?: string = 'Click Here';
  @property()
  linkTitle?: string = '';
  @property()
  href?: string = '#';
  @property()
  rel?: string = '';
  @property()
  target?: string = '_self';
  @property()
  type?: LinkComponentTypes = 'link';
  @property()
  slotName?: string = '';

  protected getClassTheme() {
    return this.type === LinkTypes.Default
      ? 'font-medium text-blue-600 dark:text-blue-500 hover:underline'
      : 'text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded text-sm px-5 py-2.5 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800 inline-block ';
  }

  render() {
    return html`
      <a
        part="custom-link"
        class="${this.getClassTheme()}"
        title=${this.linkTitle}
        href=${this.href}
        target=${this.target}
        rel=${this.rel}
        aria-label=${msg(str`Click Here to redirect`)}
      >
        ${this.text ? this.text : this.text === '' ? '' : msg(str`Click Here`)}
        <slot name=${this.slotName}></slot>
      </a>
    `;
  }
}

declare global {
  interface HTMLElementTagNameMap {
    'link-component': LinkComponent;
  }
}
