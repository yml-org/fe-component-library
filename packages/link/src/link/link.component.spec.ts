import { LinkComponent } from './link.component';
import {
  linkConfig,
  linkConfigTypeButton,
  linkConfigSlot,
} from '../constants/link.variants';

describe('ymlwebcl-link', () => {
  const LINK_COMPONENT = 'ymlwebcl-link';
  let linkElement: LinkComponent;

  const getShadowRoot = (tagName: string): ShadowRoot | null =>
    document.body.getElementsByTagName(tagName)[0].shadowRoot;

  beforeEach(() => {
    linkElement = window.document.createElement(
      LINK_COMPONENT
    ) as LinkComponent;
    document.body.appendChild(linkElement);
  });

  afterEach(() => {
    document.body.getElementsByTagName(LINK_COMPONENT)[0].remove();
  });

  it('matches component snapshot', async () => {
    expect(
      document.body.getElementsByTagName(LINK_COMPONENT)[0]
    ).toMatchSnapshot();
  });

  it('renders the component with default type', async () => {
    window.document.body.appendChild(linkElement);
    await linkElement.updateComplete;
    const elem = getShadowRoot(LINK_COMPONENT).querySelector('a');
    expect(elem.classList.contains('text-blue-600')).toBe(true);
  });

  it('renders the component with button type', async () => {
    linkElement.setAttribute('type', linkConfigTypeButton.type);
    window.document.body.appendChild(linkElement);
    await linkElement.updateComplete;
    const elem = getShadowRoot(LINK_COMPONENT).querySelector('a');
    expect(elem.classList.contains('text-white')).toBe(true);
  });

  it('sets the text property correctly', async () => {
    linkElement.setAttribute('text', linkConfig.text);
    window.document.body.appendChild(linkElement);
    await linkElement.updateComplete;
    const elem = getShadowRoot(LINK_COMPONENT).querySelector('a');
    expect(elem.innerText).toBe(linkConfig.text);
  });

  it('sets the href property correctly', async () => {
    linkElement.setAttribute('href', linkConfig.href);
    window.document.body.appendChild(linkElement);
    await linkElement.updateComplete;
    const elem = getShadowRoot(LINK_COMPONENT).querySelector('a');
    expect(elem.getAttribute('href')).toBe(linkConfig.href);
  });

  it('sets the rel property correctly', async () => {
    linkElement.setAttribute('rel', linkConfig.rel);
    window.document.body.appendChild(linkElement);
    await linkElement.updateComplete;
    const elem = getShadowRoot(LINK_COMPONENT).querySelector('a');
    expect(elem.getAttribute('rel')).toBe(linkConfig.rel);
  });

  it('sets the target property correctly', async () => {
    linkElement.setAttribute('target', linkConfig.target);
    window.document.body.appendChild(linkElement);
    await linkElement.updateComplete;
    const elem = getShadowRoot(LINK_COMPONENT).querySelector('a');
    expect(elem.getAttribute('target')).toBe(linkConfig.target);
  });

  it('renders the component passed in slot', async () => {
    linkElement.setAttribute('slotName', linkConfigSlot?.slotName);
    window.document.body.appendChild(linkElement);
    document.querySelector('ymlwebcl-link').innerHTML =
      '<p slot="bell" class="slot-test">Test</p>';
    await linkElement.updateComplete;
    const elem = document.querySelector('ymlwebcl-link').querySelector('p');
    expect(elem.classList.contains('slot-test')).toBe(true);
  });
});
