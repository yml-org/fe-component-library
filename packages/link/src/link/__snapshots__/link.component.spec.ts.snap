// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ymlwebcl-link matches component snapshot 1`] = `
v {
  "_$AL": Map {},
  "_$Do": H {
    "_$AA": <!---->,
    "_$AB": null,
    "_$AH": At {
      "_$AD": O {
        "el": <template />,
        "parts": Array [
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "class",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "title",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "href",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "target",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "rel",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "aria-label",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "index": 1,
            "type": 2,
          },
          Object {
            "ctor": [Function],
            "index": 2,
            "name": "name",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
        ],
      },
      "_$AM": [Circular],
      "_$AN": undefined,
      "u": Array [
        L {
          "_$AH": "font-medium text-blue-600 dark:text-blue-500 hover:underline",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <a
            aria-label="Click Here to redirect"
            class="font-medium text-blue-600 dark:text-blue-500 hover:underline"
            href="#"
            part="custom-link"
            rel=""
            target="_self"
            title=""
          >
            
        
            <!--?lit$3302509262$-->
            Click Here
            
        
            <slot
              name=""
            />
            
      
          </a>,
          "name": "class",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        L {
          "_$AH": "",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <a
            aria-label="Click Here to redirect"
            class="font-medium text-blue-600 dark:text-blue-500 hover:underline"
            href="#"
            part="custom-link"
            rel=""
            target="_self"
            title=""
          >
            
        
            <!--?lit$3302509262$-->
            Click Here
            
        
            <slot
              name=""
            />
            
      
          </a>,
          "name": "title",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        L {
          "_$AH": "#",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <a
            aria-label="Click Here to redirect"
            class="font-medium text-blue-600 dark:text-blue-500 hover:underline"
            href="#"
            part="custom-link"
            rel=""
            target="_self"
            title=""
          >
            
        
            <!--?lit$3302509262$-->
            Click Here
            
        
            <slot
              name=""
            />
            
      
          </a>,
          "name": "href",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        L {
          "_$AH": "_self",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <a
            aria-label="Click Here to redirect"
            class="font-medium text-blue-600 dark:text-blue-500 hover:underline"
            href="#"
            part="custom-link"
            rel=""
            target="_self"
            title=""
          >
            
        
            <!--?lit$3302509262$-->
            Click Here
            
        
            <slot
              name=""
            />
            
      
          </a>,
          "name": "target",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        L {
          "_$AH": "",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <a
            aria-label="Click Here to redirect"
            class="font-medium text-blue-600 dark:text-blue-500 hover:underline"
            href="#"
            part="custom-link"
            rel=""
            target="_self"
            title=""
          >
            
        
            <!--?lit$3302509262$-->
            Click Here
            
        
            <slot
              name=""
            />
            
      
          </a>,
          "name": "rel",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        L {
          "_$AH": "Click Here to redirect",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <a
            aria-label="Click Here to redirect"
            class="font-medium text-blue-600 dark:text-blue-500 hover:underline"
            href="#"
            part="custom-link"
            rel=""
            target="_self"
            title=""
          >
            
        
            <!--?lit$3302509262$-->
            Click Here
            
        
            <slot
              name=""
            />
            
      
          </a>,
          "name": "aria-label",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        H {
          "_$AA": <!--?lit$3302509262$-->,
          "_$AB": 
        ,
          "_$AH": "Click Here",
          "_$AM": [Circular],
          "_$AN": undefined,
          "_$Cm": true,
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 2,
        },
        L {
          "_$AH": "",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <slot
            name=""
          />,
          "name": "name",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
      ],
    },
    "_$AM": undefined,
    "_$AN": undefined,
    "_$Cm": true,
    "options": Object {
      "host": [Circular],
      "isConnected": true,
      "renderBefore": null,
    },
    "type": 2,
  },
  "_$EC": undefined,
  "_$E_": Promise {},
  "_$Ei": undefined,
  "_$El": null,
  "__href": "#",
  "__linkTitle": "",
  "__rel": "",
  "__slotName": "",
  "__target": "_self",
  "__text": "Click Here",
  "__type": "link",
  "enableUpdating": [Function],
  "hasUpdated": true,
  "isUpdatePending": false,
  "renderOptions": Object {
    "host": [Circular],
    "isConnected": true,
    "renderBefore": null,
  },
  "renderRoot": ShadowRoot {
    "_$litPart$": H {
      "_$AA": <!---->,
      "_$AB": null,
      "_$AH": At {
        "_$AD": O {
          "el": <template />,
          "parts": Array [
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "class",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "title",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "href",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "target",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "rel",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "aria-label",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "index": 1,
              "type": 2,
            },
            Object {
              "ctor": [Function],
              "index": 2,
              "name": "name",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
          ],
        },
        "_$AM": [Circular],
        "_$AN": undefined,
        "u": Array [
          L {
            "_$AH": "font-medium text-blue-600 dark:text-blue-500 hover:underline",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <a
              aria-label="Click Here to redirect"
              class="font-medium text-blue-600 dark:text-blue-500 hover:underline"
              href="#"
              part="custom-link"
              rel=""
              target="_self"
              title=""
            >
              
        
              <!--?lit$3302509262$-->
              Click Here
              
        
              <slot
                name=""
              />
              
      
            </a>,
            "name": "class",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          L {
            "_$AH": "",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <a
              aria-label="Click Here to redirect"
              class="font-medium text-blue-600 dark:text-blue-500 hover:underline"
              href="#"
              part="custom-link"
              rel=""
              target="_self"
              title=""
            >
              
        
              <!--?lit$3302509262$-->
              Click Here
              
        
              <slot
                name=""
              />
              
      
            </a>,
            "name": "title",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          L {
            "_$AH": "#",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <a
              aria-label="Click Here to redirect"
              class="font-medium text-blue-600 dark:text-blue-500 hover:underline"
              href="#"
              part="custom-link"
              rel=""
              target="_self"
              title=""
            >
              
        
              <!--?lit$3302509262$-->
              Click Here
              
        
              <slot
                name=""
              />
              
      
            </a>,
            "name": "href",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          L {
            "_$AH": "_self",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <a
              aria-label="Click Here to redirect"
              class="font-medium text-blue-600 dark:text-blue-500 hover:underline"
              href="#"
              part="custom-link"
              rel=""
              target="_self"
              title=""
            >
              
        
              <!--?lit$3302509262$-->
              Click Here
              
        
              <slot
                name=""
              />
              
      
            </a>,
            "name": "target",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          L {
            "_$AH": "",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <a
              aria-label="Click Here to redirect"
              class="font-medium text-blue-600 dark:text-blue-500 hover:underline"
              href="#"
              part="custom-link"
              rel=""
              target="_self"
              title=""
            >
              
        
              <!--?lit$3302509262$-->
              Click Here
              
        
              <slot
                name=""
              />
              
      
            </a>,
            "name": "rel",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          L {
            "_$AH": "Click Here to redirect",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <a
              aria-label="Click Here to redirect"
              class="font-medium text-blue-600 dark:text-blue-500 hover:underline"
              href="#"
              part="custom-link"
              rel=""
              target="_self"
              title=""
            >
              
        
              <!--?lit$3302509262$-->
              Click Here
              
        
              <slot
                name=""
              />
              
      
            </a>,
            "name": "aria-label",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          H {
            "_$AA": <!--?lit$3302509262$-->,
            "_$AB": 
        ,
            "_$AH": "Click Here",
            "_$AM": [Circular],
            "_$AN": undefined,
            "_$Cm": true,
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 2,
          },
          L {
            "_$AH": "",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <slot
              name=""
            />,
            "name": "name",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
        ],
      },
      "_$AM": undefined,
      "_$AN": undefined,
      "_$Cm": true,
      "options": Object {
        "host": [Circular],
        "isConnected": true,
        "renderBefore": null,
      },
      "type": 2,
    },
  },
}
`;
